plugins/tiddlywiki/filesystem/filesystemadaptor.js

```diff
@@ -69,11 +91,20 @@ Save a tiddler and invoke the callback with (err,adaptorInfo,revision)
 */
 FileSystemAdaptor.prototype.saveTiddler = function(tiddler,callback) {
 	var self = this;
-	this.getTiddlerFileInfo(tiddler,function(err,fileInfo) {
+	this.getTiddlerFileInfo(tiddler,function(err,fileInfo,options) {
 		if(err) {
 			return callback(err);
 		}
-		$tw.utils.saveTiddlerToFile(tiddler,fileInfo,callback);
+		if (options && options.fileInfo !== null) {
```

Comment:
here is a check for `options.fileInfo !== null` .... but fileInfo may not be `null`  ... IMO `options` was set to `null` ... Please check this. I'm not 100% sure, it just looks a little bit suspicious

Reply:
Mahalo. Debugging this was a pain for a little while until I figured out how to step into virtual file in VSCode+Node.js. Please see my above comments, and the newest refactor. Really appreciate the help. :)